### ISSUE TITLE
Suggestion for `lucent.optvis.render.hook_model`

### ISSUE THREAD

**Issue by alvinwan** [greentfrapp/lucent#11]
First, thanks for making this. Lifesaver. Two thoughts (Fwiw, the nested functions, higher-order functions and decorators make things a biiiiit hard to follow when debugging):

1. I initially dun goofed and didn't eval the model (even though the very example notebook I'm using from lucent does lol). Maybe the `hook_model` function could check for nonetypes and tell the user to eval, if no saved feature maps are found?
2. PyTorch module names usually use dot notation. Maybe use dots instead of underscores? Or just tell the user which feature map names are available and the user'll figure it out quickly enough

Suggested replacement for this function: https://github.com/greentfrapp/lucent/blob/a2b015ce95f29460a329f750428077bcde5e4e94/lucent/optvis/render.py#L194

```
def hook(layer):
        if layer == "input":
            out = image_f()
        elif layer == "labels":
            out = list(features.values())[-1].features
        else:
            assert layer in features, f"Invalid layer {layer}. Pick from one of {features.keys()}"  # suggestion 2 ish
            out = features[layer].features
        assert out is not None, "There are no saved feature maps. Make sure to put the model in eval mode, like so: `model.to(device).eval()`. See Lucent notebook for example."   # suggestion 1, tell user to eval
        return out
```

*I ran it on resnet18. Gorgeous and worked out of the box btw.

![download-4](https://user-images.githubusercontent.com/2068077/94094968-99e4ee00-fdd5-11ea-9f1b-829e27576617.png)
![download-5](https://user-images.githubusercontent.com/2068077/94094972-9b161b00-fdd5-11ea-8a97-ee13cfacec6b.png)
![download-6](https://user-images.githubusercontent.com/2068077/94094974-9c474800-fdd5-11ea-998e-ad22412cc41c.png)
![download-7](https://user-images.githubusercontent.com/2068077/94094977-9d787500-fdd5-11ea-9556-5cdfebfa25ba.png)

**Comment by greentfrapp** [comment]
Hey @alvinwan I'm glad to know that this helped! Thanks for the suggestions too! Good point and makes a lot of sense, I'll work them in

**Comment by alvinwan** [comment]
a732f42673cc53f65196defed3443ff6357df1e5
